### PR TITLE
remove reference of namespace istio-system

### PR DESCRIPTION
**Pull request by linsun**
**What this PR does / why we need it**:

User should be creating the rules in whatever namespace they would like... so that they can actually view them.  Else, if I don't have access to istio-system, I won't be able to view my rules.

I noticed later on in the actions section, we still have namespace reference in there.  Given default is the namespace we use in bookinfo, it makes sense start with that namespace.

I've tested this, and it works for me.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by linsun** [comment]
Hmm, ok found one issue with being namespace default, after the mixer-rule-deny-serviceaccount.yaml  is deleted, I'm still not able to get to the details...

**Comment by linsun** [comment]
ok, working with @douglas-reid and discovered the details just came back after I deleted the rule about 20 mins ago.

**Comment by cmluciano** [comment]
istio-system is also noted in https://github.com/istio/istio/blob/master/samples/bookinfo/kube/mixer-rule-ratings-ratelimit.yaml#L5 and https://github.com/istio/istio/blob/master/samples/bookinfo/kube/mixer-rule-ratings-denial.yaml#L5-L5 . Should these be updated as well?

File: samples/bookinfo/kube/mixer-rule-deny-label.yaml [hunk 1]
```diff
@@ -12,16 +11,14 @@ apiVersion: "config.istio.io/v1alpha2"
...
 metadata:
   name: denyreviewsv3
-  namespace: istio-system
 spec:
   match: destination.labels["app"] == "ratings" && source.labels["app"]=="reviews" && source.labels["version"] == "v3"
   actions:
-  - handler: denyreviewsv3handler.denier.istio-system
-    instances: [ denyreviewsv3request.checknothing.istio-system ]
+  - handler: denyreviewsv3handler.denier.default
+    instances: [ denyreviewsv3request.checknothing.default ]
```
Comment (jmuk) [inline samples/bookinfo/kube/mixer-rule-deny-label.yaml]: drive-by comment: you can omit those `.default` suffix, mixer will find out the handler and instances by filling the namespace when omitted. cc: @mandarjog

File: samples/bookinfo/kube/mixer-rule-deny-serviceaccount.yaml [hunk 2]
```diff
@@ -12,16 +11,14 @@ apiVersion: "config.istio.io/v1alpha2"
...
 metadata:
   name: denyproductpage
-  namespace: istio-system
 spec:
   match: destination.labels["app"] == "details" && source.user == "spiffe://cluster.local/ns/default/sa/bookinfo-productpage"
   actions:
-  - handler: denyproductpagehandler.denier.istio-system
-    instances: [ denyproductpagerequest.checknothing.istio-system ]
+  - handler: denyproductpagehandler.denier.default
+    instances: [ denyproductpagerequest.checknothing.default ]
```
Comment (jmuk) [inline samples/bookinfo/kube/mixer-rule-deny-serviceaccount.yaml]: ditto

Comment (linsun) [reply samples/bookinfo/kube/mixer-rule-deny-serviceaccount.yaml]: Thank you for suggesting that, @jmuk , that made user's experience easier!  I tried it and worked too!

**Comment by linsun** [comment]
@jmuk PTAL

**Comment by douglas-reid** [comment]
fwiw, writing rules in `istio-system` makes them (with the current configuration of mixer) universal. they will apply to all services regardless of namespacing. putting rules in a namespace limits their scope to that namespace. we should probably make that very explicit, so that the choice is understood.

**Comment by linsun** [comment]
good point, I think we should @douglas-reid!  I was very confused why our routing rule didn't specify namespace and suddenly when I move to mixer rules it is namespace in istio-system.

**Comment by linsun** [comment]
On 2nd thought, I think some of these policies doesn't make sense to be global, for example:  the ones that uses service account because service account is very specific to namespace:  mixer-rule-deny-serviceaccount.yaml

**Comment by jmuk** [comment]
/lgtm
Agreed. I think some of the config node _can_ be in istio-system -- like the global handler config can be bound by rules in individual namespaces. Having said that, I think this structure makes sense as our example.
